src/Instrumentation/Laravel/src/Watchers/LogWatcher.php

```diff
@@ -6,23 +6,37 @@
 
 use Illuminate\Contracts\Foundation\Application;
 use Illuminate\Log\Events\MessageLogged;
+use Illuminate\Log\LogManager;
 use OpenTelemetry\API\Trace\Span;
 use OpenTelemetry\Context\Context;
 
 class LogWatcher extends Watcher
 {
+    /** @var LogManager */
+    private LogManager $logger;
+
     /** @psalm-suppress UndefinedInterfaceMethod */
     public function register(Application $app): void
     {
         /** @phan-suppress-next-line PhanTypeArraySuspicious */
         $app['events']->listen(MessageLogged::class, [$this, 'recordLog']);
+
+        /** @phan-suppress-next-line PhanTypeArraySuspicious */
+        $this->logger = $app['log'];
     }
 
     /**
      * Record a log.
      */
     public function recordLog(MessageLogged $log): void
     {
+        $underlyingLogger = $this->logger->getLogger();
+
+        /** @phan-suppress-next-line PhanUndeclaredMethod */
+        if (method_exists($underlyingLogger, 'isHandling') && !$underlyingLogger->isHandling($log->level)) {
```

Comment:
looks like the `isHandling` is only available in the monolog - my understanding is that someone could pass a custom logger.

I'm not sure I like the `method_exists` - but maybe is the safest way of ensuring this doesn't break any code?

Reply:
Hey Alec :wave:

I had the same understanding, as it looks like PSR-3 doesn't account for that functionality.

This isn't the first `method_exists()` usage at least across contrib. Another option could be an `instanceof` check to see if the method exists and is compatible.

I wonder if OTel has some filtering mechanism in general for this, as I imagine the PSR-3 instrumentation would log everything as well.

Reply:
👋🏻  Hi! 

I do think that this isn't perfect - but maybe prevents at least some leaking of debug logs into enviroments where it shouldn't be.

Looks like a processor could also be used - i.e. in a collector - to give more options for dropping events - see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md.
However this is maybe more advanced - and for people with basic setups it's probably better to have this as a fallback.

Reply:
Is it expected that the underlying logger could change? If not, maybe it is possible to get the underlying one already in `register` and only assign to a property if `method_exists` passes there.